fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -211,4 +174,165 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
             }
         }
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        translationDialogFeature.set(
+            feature = TranslationsDialogFeature(
+                browserStore = browserStore,
+                translationsDialogStore = translationsDialogStore,
+                sessionId = args.sessionId,
+            ),
+            owner = this,
+            view = view,
+        )
+        translationsDialogStore.dispatch(TranslationsDialogAction.InitTranslationsDialog)
+    }
+
+    @Composable
+    private fun TranslationsDialogContent(learnMoreUrl: String, onSettingClicked: () -> Unit) {
+        val fromSelected =
+            translationsDialogStore.observeAsComposableState { state -> state.initialFrom }.value
+
+        val toSelected =
+            translationsDialogStore.observeAsComposableState { state -> state.initialTo }.value
+
+        setTranslatedContentDescription(fromSelected, toSelected)
+
+        val translateFromLanguages =
+            translationsDialogStore.observeAsComposableState { state -> state.fromLanguages }.value
+
+        val translateToLanguages = translationsDialogStore.observeAsComposableState { state ->
+            state.toLanguages
+        }.value
+
+        val positiveButtonType = translationsDialogStore.observeAsComposableState { state ->
+            state.positiveButtonType
+        }.value
+
+        val dismissDialogState = translationsDialogStore.observeAsComposableState { state ->
+            state.dismissDialogState
+        }.value
+
+        val translationError = translationsDialogStore.observeAsComposableState { state ->
+            state.error
+        }.value
+
+        isInProgress = translationsDialogStore.observeAsComposableState { state ->
+            state.isInProgress
+        }.value
+
+        if (dismissDialogState is DismissDialogState.Dismiss) {
+            dismissDialog()
+        }
+
+        val isTranslated = translationsDialogStore.observeAsComposableState { state ->
+            state.isTranslated
+        }.value
+
+        setTranslatedPageTitle(fromSelected = fromSelected, toSelected = toSelected)
+
+        TranslationsDialog(
+            translationsDialogState = TranslationsDialogState(
```

Comment:
Do we need all these observations just to recreate a `TranslationsDialogState`, or can we just observe the `translationsDialogStore.state` directly and pass that down?

Reply:
Done